JavaScript/JavaScript.sublime-syntax

```diff
@@ -164,12 +166,46 @@ contexts:
       push: jsdoc-block-tag
 
   jsdoc-block-tag:
+    - match: '{{jsdoc_block_tag_with_type}}'
+      scope: entity.other.attribute-name.documentation.js
+      set:
+        - match: '\s+(?=\{)'
+          set: jsdoc-type-expression
+        - match: ''
+          pop: 1
     - match: '{{jsdoc_block_tag}}'
       scope: entity.other.attribute-name.documentation.js
       pop: 1
     - match: (?=\S)|$
       pop: 1
 
+  jsdoc-type-expression:
+    - match: '\{'
+      scope: punctuation.definition.type.begin.js, meta.tag.jsdoc
+      set:
+        - match: '\}'
+          scope: punctuation.definition.type.end.js, meta.tag.jsdoc
+          pop: 1
+        - include: jsdoc-type-expression-body
+
+  jsdoc-type-expression-body:
+    - match: (?=\*/)
+      pop: 1
+    - match: '^\s*\*'
+    - match: \.
+      scope: meta.type.js punctuation.accessor.js
+    - match: '\{'
+      scope: meta.type.js
```

Comment:
`punctuation.section.block` or `punctuation.definition.block`?

And also, this would apply a distinct color. At least in Mariana the first would become white and the second blue. So it would be a bit in conflict with my latest change.

Reply:
but aren't you already scoping `{` and `}` in type comments? https://github.com/sublimehq/Packages/pull/2629/files#diff-04a9b98e8498a3df8d16447d91d8c3cb9131fa4359f113b2917651aa0ed77c23R183-R184

Reply:
I would guess `section` as I don't really see `definition` listed on http://www.sublimetext.com/docs/scope_naming.html#punctuation regarding curly braces

Reply:
> but aren't you already scoping `{` and `}` in type comments? [#2629 (files)](https://github.com/sublimehq/Packages/pull/2629/files#diff-04a9b98e8498a3df8d16447d91d8c3cb9131fa4359f113b2917651aa0ed77c23R183-R184)

Yes but with the same scope as the rest of the type contents.

The outer braces are scoped distinctly but those can be see as part of the jsdoc tag syntax rather than the type itself.

![Screenshot 2023-05-14 at 11 07 08](https://github.com/sublimehq/Packages/assets/153197/dcdb167e-0583-4347-8c1f-a67dd7dc1fb0)

Reply:
ah I see, thanks for the explanation. Then feel free to ignore this suggestion :)